packages/auth/test/integration/flows/totp.test.ts

```diff
@@ -0,0 +1,147 @@
+              verificationCodeLength: 30,
+              hashingAlgorithm: 'sha1',
+              periodSec:30,
+              sessionInfo: 'testsSessionInfo',
+              finalizeEnrollmentTime:  Date.now()
+              },
+    code: '...'
+  };
+  
+  const TOTP_COMB_B = {
+
+    response: { sharedSecretKey: 'secretKey2',
+    verificationCodeLength: 30,
+    hashingAlgorithm: 'sha1',
+    periodSec: 30,
+    sessionInfo: 'testsSessionInfo',
+    finalizeEnrollmentTime:  Date.now()
+    },
+    code: '...'
+  };
+
+describe(' Integration tests: Mfa TOTP', () => {
+    let auth: Auth;
+    let idToken: string;
+    let signUpCred: UserCredential;
+    let email: string;
+    let assertion: MultiFactorAssertionImpl;
+    let _request: MFA.StartTotpMfaEnrollmentRequest;
+    let startMfaResponse: MFA.StartTotpMfaEnrollmentResponse;
+    let displayName: string;
+  beforeEach(async () => {
+    auth = getTestInstance();
+    email =randomEmail();
+    idToken = 'testIdToken';
+    signUpCred = await createUserWithEmailAndPassword(
+      auth,
+      email,
+      'password'
+    );
+    await auth.signOut();
+  });
+ 
+  afterEach(async () => {
+    await cleanUpTestInstance(auth);
+
+  });
+    it('should verify using otp', async () => {
+
+      console.log(email);
+      const cr = await signInWithEmailAndPassword(auth, email, 'password');
+
+      startMfaResponse = { totpSessionInfo: TOTP_COMB_A.response}
+      
+
+     
+      const mfaUser =  multiFactor(cr.user);
+      sinon.spy(MultiFactorSessionImpl, '_fromIdtoken');
+ 
+      sinon.stub(mfaUser, 'getSession').returns(
+       Promise.resolve(MultiFactorSessionImpl._fromIdtoken(idToken, auth as any)));
+        
+      sinon.stub(MFA, 'startEnrollTotpMfa').callsFake((_auth,_request)=>{
+
+        return Promise.resolve(startMfaResponse)
+      })
+      
+     
+
+      const session =  await mfaUser.getSession();
+       
+      console.log(session);
+      
+      const totpSecret = await TotpMultiFactorGenerator.generateSecret(
+        session
+      );
+
+      console.log("**** totpSecret"+ totpSecret);
+      // https://stackoverflow.com/questions/48931815/sinon-stub-not-replacing-function
+      // https://stackoverflow.com/questions/61051247/chai-spies-expect-to-have-been-called-is-failing-on-local-methods
+      expect(MultiFactorSessionImpl._fromIdtoken).to.have.been.calledOnce;
+      //expect(TotpSecret._fromStartTotpMfaEnrollmentResponse).to.have.been.calledOnce;
+      expect(MFA.startEnrollTotpMfa).to.have.been.calledOnce;
+
+      expect(await MFA.startEnrollTotpMfa(auth as any, _request)).to.eql(startMfaResponse)
+      
+      expect(totpSecret.secretKey).to.eql(startMfaResponse.totpSessionInfo.sharedSecretKey)
+      expect(totpSecret.codeLength).to.eql(startMfaResponse.totpSessionInfo.verificationCodeLength)
+
+      const totpVerificationCode = await mockTotp(totpSecret.secretKey, totpSecret.codeLength, totpSecret.codeIntervalSeconds);
```

Comment:
nit: instead of mockTotp, we can call it "getTotpCode" since it generates a real code for the given params.